### DISCUSSION TITLE
rabbitmqctl encode accepts plain text values

### DISCUSSION THREAD

**Discussion by ohye3166** [rabbitmq/rabbitmq-server#4216] [Other] [answered]
![image](https://user-images.githubusercontent.com/60819151/156287623-7ab2a702-051a-4e16-aa67-6cc898cf611a.png)
ps -ef | grep rabbitmq will show Plaintext password when using rabbitmqctl encode function which is unsafe, so if there support Interactive way to input password?

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment, marked as answer]
CLI commands that operate on passwords do accept them via standard input. `rabbitmqctl encode` does not.

It is not meant to be used on production nodes. Used it to encode a value and use whatever configuration generation tooling you use
to deploy it.

**Comment by michaelklishin** [comment]
Curiously, both the passphrase and the value encrypted can in practice be considered sensitive. So which one should be accepted via standard input? Or both? I wonder if it would have any unintended consequences. Doing that for the passphrase would be trivial, as it is something that `change_password` already does. We don't have any examples of commands that accept all positional arguments via standard input.

**Comment by michaelklishin** [comment]
See https://github.com/rabbitmq/rabbitmq-server/pull/4258.
